gcc/rust/parse/rust-parse-impl.h

```diff
@@ -4922,8 +4922,8 @@ Parser<ManagedTokenSource>::parse_trait (AST::Visibility vis,
 
   if (is_auto_trait && !trait_items.empty ())
     {
-      add_error (
-	Error (locus, "associated items are forbidden within auto traits"));
+      rust_error_at (locus, ErrorCode::E0380,
+		     "auto traits cannot have methods or associated items");
```

Comment:
@philberty Parsing error needs to be collected using `Error` struct. I am working on this issue

- https://github.com/Rust-GCC/gccrs/issues/2385

Once, this issue get fixed. I'll then use the `Error` Struct to pass the `errorcode` instead of `rust_error_at`. If this doesn't change the behaviour of the code, then I am happy to remove this from draft.